hopsworks-IT/src/test/ruby/spec/kafka_acl_spec.rb

```diff
@@ -0,0 +1,501 @@
+ Copyright (C) 2019, Logical Clocks AB. All rights reserved
+
+ Hopsworks is free software: you can redistribute it and/or modify it under the terms of
+ the GNU Affero General Public License as published by the Free Software Foundation,
+ either version 3 of the License, or (at your option) any later version.
+
+ Hopsworks is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+ PURPOSE.  See the GNU Affero General Public License for more details.
+
+ You should have received a copy of the GNU Affero General Public License along with this program.
+ If not, see <https://www.gnu.org/licenses/>.
+=end
+
+require 'json'
+
+describe "On #{ENV['OS']}" do
+  describe 'kafka acl' do
+    after (:all) { clean_projects }
+
+    describe 'basic acl operations' do
+      context 'with valid project with two users, kafka service enabled, and a kafka schema' do
+        before :all do
+          with_valid_project
+          project = get_project
+          new_member = create_user[:email]
+          post "#{ENV['HOPSWORKS_API']}/project/#{@project[:id]}/projectMembers", {projectTeam: [{projectTeamPK: {projectId: @project[:id],teamMember: new_member},teamRole: "Data scientist"}]}
+          expect_json(successMessage: "One member added successfully")
+          expect_status(200)
+          with_kafka_schema(project.id)
+          with_kafka_topic(project.id)
+        end
+        after :all do
+          project = get_project
+          clean_topics(project.id) 
+        end
+        it 'should add acls for all users of the project for a new kafka topic' do
+          project = get_project
+          topic_name = get_topic
+          get "#{ENV['HOPSWORKS_API']}/project/#{@project[:id]}/projectMembers"
+          usersInProject = json_body.size
+          get_kafka_acls(project, topic_name)
+          expect_status(200)
+          expect(json_body[:items].count).to eq(usersInProject) 
+        end
+        it 'should add a new acl after adding a new member of the project' do
+          get "#{ENV['HOPSWORKS_API']}/project/#{@project[:id]}/projectMembers"
+          usersInProject = json_body.size 
+          new_member = create_user[:email]
+          post "#{ENV['HOPSWORKS_API']}/project/#{@project[:id]}/projectMembers", {projectTeam: [{projectTeamPK: {projectId: @project[:id],teamMember: new_member},teamRole: "Data scientist"}]}
+          get "#{ENV['HOPSWORKS_API']}/project/#{@project[:id]}/projectMembers"
+          usersInProjectUpdated = json_body.size
+          expect(usersInProjectUpdated).to eq(usersInProject + 1)
+          project = get_project
+          topic_name = get_topic
+          get_kafka_acls(project, topic_name)
+          expect_status(200)
+          expect(json_body[:items].count).to eq(usersInProjectUpdated)
+        end
+        it 'should remove all acls related to a member after removing this member from a project' do
```

Comment:
@kai-chi how about when removing a member from a project the topic has been shared with?

Reply:
added a new test